File: md5.java
import com.google.gson.JsonObject;

public class Main {
    public static JsonObject main(JsonObject args) {
        // if a value for name is provided, use it else use a default
        String input = args.has("name") ? args.get("name").getAsString() : "stranger";

        int output = hash(input);

        // construct the response JSON object
        JsonObject response = new JsonObject();
        response.addProperty("result", output);
        response.addProperty("latency", 1);

        return response;
    }

    public static int hash(String input) {
        int hash = 0;

        for (int i = 0; i < input.length(); i++) {
            hash = (hash << 5) - hash + input.charAt(i);
            hash &= hash;
        }

        return hash;
    }
}

File: md5.php
function hash($input) {
  $hash = 0;

  for ($i = 0; $i < strlen($input); $i++) {
    $hash = ($hash << 5) - $hash + ord($input[$i]);
    $hash &= $hash;
  }

  return $hash;
}

function main(array $params) : array {
  // if a value for name is provided, use it else use a default
  $input = isset($params['name']) ? $params['name'] : 'stranger';

  $output = hash($input);

  // construct the message using the values for name and place
  return [
    'result' => $output,
    'latency' => 1
  ];
}
